presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java

```diff
@@ -0,0 +1,176 @@
+package com.facebook.presto.hive;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.airlift.log.Logging;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.tpch.TpchTable.NATION;
+import static java.lang.String.format;
+
+public class HiveExternalWorkerQueryRunner
+{
+    private static final Logger log = Logger.get(HiveQueryRunner.class);
+
+    public static QueryRunner createQueryRunner()
+            throws Exception
+    {
+        String prestoServerPath = System.getenv("PRESTO_SERVER");
+        String baseDataDir = System.getenv("DATA_DIR");
+        String workerCount = System.getenv("WORKER_COUNT");
+
+        return createQueryRunner(Optional.ofNullable(prestoServerPath),
+                Optional.ofNullable(baseDataDir).map(Paths::get),
+                Optional.ofNullable(workerCount).map(Integer::parseInt));
+    }
+
+    public static DistributedQueryRunner createQueryRunner(Optional<String> prestoServerPath,
+            Optional<Path> baseDataDir,
+            Optional<Integer> workerCount)
+            throws Exception
+    {
+        if (prestoServerPath.isPresent()) {
+            checkArgument(baseDataDir.isPresent(), "Path to data files must be specified when testing external workers");
+        }
+
+        DistributedQueryRunner defaultQueryRunner = HiveQueryRunner.createQueryRunner(
+                ImmutableList.of(NATION),
+                ImmutableMap.of(),
+                "sql-standard",
+                ImmutableMap.of("hive.storage-format", "DWRF"),
+                baseDataDir);
+
+        // DWRF doesn't support date type. Convert date columns to varchar for lineitem and orders.
+        createLineitem(defaultQueryRunner);
+        createOrders(defaultQueryRunner);
+
+        if (!prestoServerPath.isPresent()) {
+            return defaultQueryRunner;
+        }
+
+        defaultQueryRunner.close();
+
+        // Make query runner with external workers for tests
+        DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(ImmutableList.of(),
```

Comment:
nit: not need to create another variable `DistributedQueryRunner queryRunner = ...`. Directly return the result.